pinot-common/src/main/java/org/apache/pinot/common/utils/CommonConstants.java

```diff
@@ -386,6 +388,12 @@ public ServerType getServerType() {
       public static String REPLICA_GROUP_SEGMENT_ASSIGNMENT_STRATEGY = "ReplicaGroupSegmentAssignmentStrategy";
     }
 
+    public static class VirtualColumn {
```

Comment:
```suggestion
    public static class BuiltInVirtualColumns {
```

Reply:
IIRC you can add a virtual column in the schema with its own provider.
I am not sure if there is a restriction that virtual columns have to start with the "$" sign, but we can discuss if we can introduce that now.